Sources/ViewModels/ALKConversationViewModel.swift

```diff
@@ -795,6 +798,16 @@ open class ALKConversationViewModel: NSObject, Localizable {
             }
             updateUser(kmField: kmField, message: alMessage.message)
         }
+        
+        if emailCollectionAwayModeEnabled {
+            if(!message.isValidEmail()) {
+                delegate?.isEmailUpdatedforUser(status: false)
+                return
+            }
+            emailCollectionAwayModeEnabled = false
+            updateEmailFromCollectEmail(email: message)
+            delegate?.isEmailUpdatedforUser(status: true)
+        }
```

Comment:
I understand that using a callback would be a good option, but in this scenario, we also need to send a message and perform multiple tasks while updating the email. The following code needs to execute without delay. In case of failure, The update email UI will again come as there is another check implemented in different part of code.